node/test/builder.go

```diff
@@ -276,12 +280,24 @@ func mockBuilderOpts(t *testing.T, fullOpts []test.FullNodeOpts, storage []test.
 		maddrs = append(maddrs, maddr)
 		genms = append(genms, *genm)
 	}
+
+	var vrk genesis.Actor
+	if rkhKey != nil {
+		vrk = genesis.Actor{
+			Type:    genesis.TAccount,
+			Balance: big.Mul(big.NewInt(400000000), types.NewInt(build.FilecoinPrecision)),
+			Meta:    (&genesis.AccountMeta{Owner: rkhKey.Address}).ActorMeta(),
+		}
+	} else {
+		vrk = gen.DefaultVerifregRootkeyActor
+	}
+
 	templ := &genesis.Template{
 		Accounts:         genaccs,
 		Miners:           genms,
 		NetworkName:      "test",
 		Timestamp:        uint64(time.Now().Unix() - 10000), // some time sufficiently far in the past
-		VerifregRootKey:  gen.DefaultVerifregRootkeyActor,
+		VerifregRootKey:  vrk,
```

Comment:
We could just make this the first miner key (`genaccs[0]`), this way we don't need the separate constructors